File: CSharp/ILinkai.Weixin.Sdk/Models/ApiModels.cs
using System.Text.Json.Serialization;

namespace ILinkai.Weixin.Sdk.Models;

/// <summary>
/// 获取上传URL请求
/// </summary>
public class GetUploadUrlRequest
{
    /// <summary>
    /// 文件键
    /// </summary>
    [JsonPropertyName("filekey")]
    public string? FileKey { get; set; }

    /// <summary>
    /// 媒体类型
    /// </summary>
    [JsonPropertyName("media_type")]
    public int? MediaType { get; set; }

    /// <summary>
    /// 目标用户ID
    /// </summary>
    [JsonPropertyName("to_user_id")]
    public string? ToUserId { get; set; }

    /// <summary>
    /// 原文件明文大小
    /// </summary>
    [JsonPropertyName("rawsize")]
    public int? RawSize { get; set; }

    /// <summary>
    /// 原文件明文MD5
    /// </summary>
    [JsonPropertyName("rawfilemd5")]
    public string? RawFileMd5 { get; set; }

    /// <summary>
    /// 原文件密文大小（AES-128-ECB加密后）
    /// </summary>
    [JsonPropertyName("filesize")]
    public int? FileSize { get; set; }

    /// <summary>
    /// 缩略图明文大小（IMAGE/VIDEO时必填）
    /// </summary>
    [JsonPropertyName("thumb_rawsize")]
    public int? ThumbRawSize { get; set; }

    /// <summary>
    /// 缩略图明文MD5（IMAGE/VIDEO时必填）
    /// </summary>
    [JsonPropertyName("thumb_rawfilemd5")]
    public string? ThumbRawFileMd5 { get; set; }

    /// <summary>
    /// 缩略图密文大小（IMAGE/VIDEO时必填）
    /// </summary>
    [JsonPropertyName("thumb_filesize")]
    public int? ThumbFileSize { get; set; }

    /// <summary>
    /// 不需要缩略图上传URL，默认false
    /// </summary>
    [JsonPropertyName("no_need_thumb")]
    public bool? NoNeedThumb { get; set; }

    /// <summary>
    /// 加密密钥
    /// </summary>
    [JsonPropertyName("aeskey")]
    public string? AesKey { get; set; }

    /// <summary>
    /// 基础信息
    /// </summary>
    [JsonPropertyName("base_info")]
    public BaseInfo? BaseInfo { get; set; }
}

/// <summary>
/// 获取上传URL响应
/// </summary>
public class GetUploadUrlResponse
{
    /// <summary>
    /// 原图上传加密参数
    /// </summary>
    [JsonPropertyName("upload_param")]
    public string? UploadParam { get; set; }

    /// <summary>
    /// 缩略图上传加密参数，无缩略图时为空
    /// </summary>
    [JsonPropertyName("thumb_upload_param")]
    public string? ThumbUploadParam { get; set; }
}

/// <summary>
/// 获取更新请求
/// </summary>
public class GetUpdatesRequest
{
    /// <summary>
    /// 同步缓冲区（已弃用）
    /// </summary>
    [JsonPropertyName("sync_buf")]
    [Obsolete("已弃用，请使用GetUpdatesBuf")]
    public string? SyncBuf { get; set; }

    /// <summary>
    /// 完整上下文缓冲区，本地缓存；无缓存时发送空字符串
    /// </summary>
    [JsonPropertyName("get_updates_buf")]
    public string? GetUpdatesBuf { get; set; }

    /// <summary>
    /// 基础信息
    /// </summary>
    [JsonPropertyName("base_info")]
    public BaseInfo? BaseInfo { get; set; }
}

/// <summary>
/// 获取更新响应
/// </summary>
public class GetUpdatesResponse
{
    /// <summary>
    /// 返回码
    /// </summary>
    [JsonPropertyName("ret")]
    public int? Ret { get; set; }

    /// <summary>
    /// 错误码（例如-14表示会话超时）
    /// </summary>
    [JsonPropertyName("errcode")]
    public int? ErrCode { get; set; }

    /// <summary>
    /// 错误消息
    /// </summary>
    [JsonPropertyName("errmsg")]
    public string? ErrMsg { get; set; }

    /// <summary>
    /// 消息列表
    /// </summary>
    [JsonPropertyName("msgs")]
    public List<WeixinMessage>? Messages { get; set; }

    /// <summary>
    /// 同步缓冲区（已弃用）
    /// </summary>
    [JsonPropertyName("sync_buf")]
    [Obsolete("已弃用，请使用GetUpdatesBuf")]
    public string? SyncBuf { get; set; }

    /// <summary>
    /// 完整上下文缓冲区，需本地缓存并在下次请求时发送
    /// </summary>
    [JsonPropertyName("get_updates_buf")]
    public string? GetUpdatesBuf { get; set; }

    /// <summary>
    /// 服务器建议的下一次长轮询超时时间(毫秒)
    /// </summary>
    [JsonPropertyName("longpolling_timeout_ms")]
    public int? LongPollingTimeoutMs { get; set; }
}

/// <summary>
/// 发送消息请求
/// </summary>
public class SendMessageRequest
{
    /// <summary>
    /// 消息内容
    /// </summary>
    [JsonPropertyName("msg")]
    public WeixinMessage? Message { get; set; }
}

/// <summary>
/// 发送消息响应
/// </summary>
public class SendMessageResponse
{
}

/// <summary>
/// 发送输入状态请求
/// </summary>
public class SendTypingRequest
{
    /// <summary>
    /// ILink用户ID
    /// </summary>
    [JsonPropertyName("ilink_user_id")]
    public string? ILinkUserId { get; set; }

    /// <summary>
    /// 输入票据
    /// </summary>
    [JsonPropertyName("typing_ticket")]
    public string? TypingTicket { get; set; }

    /// <summary>
    /// 状态：1=正在输入（默认），2=取消输入
    /// </summary>
    [JsonPropertyName("status")]
    public int? Status { get; set; }

    /// <summary>
    /// 基础信息
    /// </summary>
    [JsonPropertyName("base_info")]
    public BaseInfo? BaseInfo { get; set; }
}

/// <summary>
/// 发送输入状态响应
/// </summary>
public class SendTypingResponse
{
    /// <summary>
    /// 返回码
    /// </summary>
    [JsonPropertyName("ret")]
    public int? Ret { get; set; }

    /// <summary>
    /// 错误消息
    /// </summary>
    [JsonPropertyName("errmsg")]
    public string? ErrMsg { get; set; }
}

/// <summary>
/// 获取配置请求
/// </summary>
public class GetConfigRequest
{
    /// <summary>
    /// ILink用户ID
    /// </summary>
    [JsonPropertyName("ilink_user_id")]
    public string? ILinkUserId { get; set; }

    /// <summary>
    /// 上下文令牌
    /// </summary>
    [JsonPropertyName("context_token")]
    public string? ContextToken { get; set; }

    /// <summary>
    /// 基础信息
    /// </summary>
    [JsonPropertyName("base_info")]
    public BaseInfo? BaseInfo { get; set; }
}

/// <summary>
/// 获取配置响应
/// </summary>
public class GetConfigResponse
{
    /// <summary>
    /// 返回码
    /// </summary>
    [JsonPropertyName("ret")]
    public int? Ret { get; set; }

    /// <summary>
    /// 错误消息
    /// </summary>
    [JsonPropertyName("errmsg")]
    public string? ErrMsg { get; set; }

    /// <summary>
    /// Base64编码的输入票据，用于sendTyping
    /// </summary>
    [JsonPropertyName("typing_ticket")]
    public string? TypingTicket { get; set; }
}

/// <summary>
/// 获取二维码响应
/// </summary>
public class GetQRCodeResponse
{
    /// <summary>
    /// 二维码标识
    /// </summary>
    [JsonPropertyName("qrcode")]
    public string? QRCode { get; set; }

    /// <summary>
    /// 二维码图片内容（URL）
    /// </summary>
    [JsonPropertyName("qrcode_img_content")]
    public string? QRCodeImgContent { get; set; }
}

/// <summary>
/// 获取二维码状态响应
/// </summary>
public class GetQRCodeStatusResponse
{
    /// <summary>
    /// 状态：wait/scaned/confirmed/expired
    /// </summary>
    [JsonPropertyName("status")]
    public string? Status { get; set; }

    /// <summary>
    /// 机器人令牌
    /// </summary>
    [JsonPropertyName("bot_token")]
    public string? BotToken { get; set; }

    /// <summary>
    /// ILink机器人ID
    /// </summary>
    [JsonPropertyName("ilink_bot_id")]
    public string? ILinkBotId { get; set; }

    /// <summary>
    /// 基础URL
    /// </summary>
    [JsonPropertyName("baseurl")]
    public string? BaseUrl { get; set; }

    /// <summary>
    /// 扫码用户的用户ID
    /// </summary>
    [JsonPropertyName("ilink_user_id")]
    public string? ILinkUserId { get; set; }
}

File: Java/src/main/java/com/ilinkai/weixin/models/ApiModels.java
package com.ilinkai.weixin.models;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/** API请求/响应模型集合 */
public class ApiModels {

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetUploadUrlRequest {
        @JsonProperty("filekey") private String fileKey;
        @JsonProperty("media_type") private Integer mediaType;
        @JsonProperty("to_user_id") private String toUserId;
        @JsonProperty("rawsize") private Integer rawSize;
        @JsonProperty("rawfilemd5") private String rawFileMd5;
        @JsonProperty("filesize") private Integer fileSize;
        @JsonProperty("thumb_rawsize") private Integer thumbRawSize;
        @JsonProperty("thumb_rawfilemd5") private String thumbRawFileMd5;
        @JsonProperty("thumb_filesize") private Integer thumbFileSize;
        @JsonProperty("no_need_thumb") private Boolean noNeedThumb;
        @JsonProperty("aeskey") private String aesKey;
        @JsonProperty("base_info") private BaseInfo baseInfo;

        public String getFileKey() { return fileKey; }
        public void setFileKey(String v) { this.fileKey = v; }
        public Integer getMediaType() { return mediaType; }
        public void setMediaType(Integer v) { this.mediaType = v; }
        public String getToUserId() { return toUserId; }
        public void setToUserId(String v) { this.toUserId = v; }
        public Integer getRawSize() { return rawSize; }
        public void setRawSize(Integer v) { this.rawSize = v; }
        public String getRawFileMd5() { return rawFileMd5; }
        public void setRawFileMd5(String v) { this.rawFileMd5 = v; }
        public Integer getFileSize() { return fileSize; }
        public void setFileSize(Integer v) { this.fileSize = v; }
        public Integer getThumbRawSize() { return thumbRawSize; }
        public void setThumbRawSize(Integer v) { this.thumbRawSize = v; }
        public String getThumbRawFileMd5() { return thumbRawFileMd5; }
        public void setThumbRawFileMd5(String v) { this.thumbRawFileMd5 = v; }
        public Integer getThumbFileSize() { return thumbFileSize; }
        public void setThumbFileSize(Integer v) { this.thumbFileSize = v; }
        public Boolean getNoNeedThumb() { return noNeedThumb; }
        public void setNoNeedThumb(Boolean v) { this.noNeedThumb = v; }
        public String getAesKey() { return aesKey; }
        public void setAesKey(String v) { this.aesKey = v; }
        public BaseInfo getBaseInfo() { return baseInfo; }
        public void setBaseInfo(BaseInfo v) { this.baseInfo = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetUploadUrlResponse {
        @JsonProperty("upload_param") private String uploadParam;
        @JsonProperty("upload_full_url") private String uploadFullUrl;
        @JsonProperty("thumb_upload_param") private String thumbUploadParam;
        @JsonProperty("thumb_upload_full_url") private String thumbUploadFullUrl;
        public String getUploadParam() { return uploadParam; }
        public void setUploadParam(String v) { this.uploadParam = v; }
        public String getUploadFullUrl() { return uploadFullUrl; }
        public void setUploadFullUrl(String v) { this.uploadFullUrl = v; }
        public String getThumbUploadParam() { return thumbUploadParam; }
        public void setThumbUploadParam(String v) { this.thumbUploadParam = v; }
        public String getThumbUploadFullUrl() { return thumbUploadFullUrl; }
        public void setThumbUploadFullUrl(String v) { this.thumbUploadFullUrl = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetUpdatesRequest {
        @JsonProperty("get_updates_buf") private String getUpdatesBuf;
        @JsonProperty("base_info") private BaseInfo baseInfo;
        public String getGetUpdatesBuf() { return getUpdatesBuf; }
        public void setGetUpdatesBuf(String v) { this.getUpdatesBuf = v; }
        public BaseInfo getBaseInfo() { return baseInfo; }
        public void setBaseInfo(BaseInfo v) { this.baseInfo = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetUpdatesResponse {
        @JsonProperty("ret") private Integer ret;
        @JsonProperty("errcode") private Integer errCode;
        @JsonProperty("errmsg") private String errMsg;
        @JsonProperty("msgs") private List<WeixinMessage> messages;
        @JsonProperty("get_updates_buf") private String getUpdatesBuf;
        @JsonProperty("longpolling_timeout_ms") private Integer longPollingTimeoutMs;

        public Integer getRet() { return ret; }
        public void setRet(Integer v) { this.ret = v; }
        public Integer getErrCode() { return errCode; }
        public void setErrCode(Integer v) { this.errCode = v; }
        public String getErrMsg() { return errMsg; }
        public void setErrMsg(String v) { this.errMsg = v; }
        public List<WeixinMessage> getMessages() { return messages; }
        public void setMessages(List<WeixinMessage> v) { this.messages = v; }
        public String getGetUpdatesBuf() { return getUpdatesBuf; }
        public void setGetUpdatesBuf(String v) { this.getUpdatesBuf = v; }
        public Integer getLongPollingTimeoutMs() { return longPollingTimeoutMs; }
        public void setLongPollingTimeoutMs(Integer v) { this.longPollingTimeoutMs = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class SendMessageRequest {
        @JsonProperty("msg") private WeixinMessage message;
        public WeixinMessage getMessage() { return message; }
        public void setMessage(WeixinMessage v) { this.message = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class SendTypingRequest {
        @JsonProperty("ilink_user_id") private String iLinkUserId;
        @JsonProperty("typing_ticket") private String typingTicket;
        @JsonProperty("status") private Integer status;
        @JsonProperty("base_info") private BaseInfo baseInfo;

        public String getILinkUserId() { return iLinkUserId; }
        public void setILinkUserId(String v) { this.iLinkUserId = v; }
        public String getTypingTicket() { return typingTicket; }
        public void setTypingTicket(String v) { this.typingTicket = v; }
        public Integer getStatus() { return status; }
        public void setStatus(Integer v) { this.status = v; }
        public BaseInfo getBaseInfo() { return baseInfo; }
        public void setBaseInfo(BaseInfo v) { this.baseInfo = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class SendTypingResponse {
        @JsonProperty("ret") private Integer ret;
        @JsonProperty("errmsg") private String errMsg;
        public Integer getRet() { return ret; }
        public void setRet(Integer v) { this.ret = v; }
        public String getErrMsg() { return errMsg; }
        public void setErrMsg(String v) { this.errMsg = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetConfigRequest {
        @JsonProperty("ilink_user_id") private String iLinkUserId;
        @JsonProperty("context_token") private String contextToken;
        @JsonProperty("base_info") private BaseInfo baseInfo;

        public String getILinkUserId() { return iLinkUserId; }
        public void setILinkUserId(String v) { this.iLinkUserId = v; }
        public String getContextToken() { return contextToken; }
        public void setContextToken(String v) { this.contextToken = v; }
        public BaseInfo getBaseInfo() { return baseInfo; }
        public void setBaseInfo(BaseInfo v) { this.baseInfo = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetConfigResponse {
        @JsonProperty("ret") private Integer ret;
        @JsonProperty("errmsg") private String errMsg;
        @JsonProperty("typing_ticket") private String typingTicket;
        public Integer getRet() { return ret; }
        public void setRet(Integer v) { this.ret = v; }
        public String getErrMsg() { return errMsg; }
        public void setErrMsg(String v) { this.errMsg = v; }
        public String getTypingTicket() { return typingTicket; }
        public void setTypingTicket(String v) { this.typingTicket = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetQRCodeResponse {
        @JsonProperty("qrcode") private String qrCode;
        @JsonProperty("qrcode_img_content") private String qrCodeImgContent;
        public String getQrCode() { return qrCode; }
        public void setQrCode(String v) { this.qrCode = v; }
        public String getQrCodeImgContent() { return qrCodeImgContent; }
        public void setQrCodeImgContent(String v) { this.qrCodeImgContent = v; }
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class GetQRCodeStatusResponse {
        @JsonProperty("status") private String status;
        @JsonProperty("bot_token") private String botToken;
        @JsonProperty("ilink_bot_id") private String iLinkBotId;
        @JsonProperty("baseurl") private String baseUrl;
        @JsonProperty("ilink_user_id") private String iLinkUserId;

        public String getStatus() { return status; }
        public void setStatus(String v) { this.status = v; }
        public String getBotToken() { return botToken; }
        public void setBotToken(String v) { this.botToken = v; }
        public String getILinkBotId() { return iLinkBotId; }
        public void setILinkBotId(String v) { this.iLinkBotId = v; }
        public String getBaseUrl() { return baseUrl; }
        public void setBaseUrl(String v) { this.baseUrl = v; }
        public String getILinkUserId() { return iLinkUserId; }
        public void setILinkUserId(String v) { this.iLinkUserId = v; }
    }
}
